### ISSUE TITLE
go-runtime seemingly failed to extract 1/2 of the modules

### ISSUE THREAD

**Issue by alecthomas** [TBD54566975/ftl#1580]
```
=== FAIL: buildengine TestGeneratedTypeRegistry (1.19s)
info:other: Building module
debug:other: Generating external modules
debug:other: Extracting schema
debug:other: Generating main module
debug:other: Tidying go.mod files
trace:other: exec: cd /Users/alec/dev/ftl/buildengine/testdata/projects/other/_ftl/go/modules && go mod tidy
trace:other: exec: cd /Users/alec/dev/ftl/buildengine/testdata/projects/other && go mod tidy
trace:other: exec: cd /Users/alec/dev/ftl/buildengine/testdata/projects/other/_ftl/go/main && go mod tidy
debug:other: go: found github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/ftlv1connect in github.com/TBD54566975/ftl v0.0.0-00010101000000-000000000000
debug:other: go: found github.com/TBD54566975/ftl/common/plugin in github.com/TBD54566975/ftl v0.0.0-00010101000000-000000000000
debug:other: go: found github.com/TBD54566975/ftl/go-runtime/ftl/reflection in github.com/TBD54566975/ftl v0.0.0-00010101000000-000000000000
debug:other: go: found github.com/TBD54566975/ftl/go-runtime/server in github.com/TBD54566975/ftl v0.0.0-00010101000000-000000000000
debug:other: go: found github.com/TBD54566975/ftl/go-runtime/ftl/reflection in github.com/TBD54566975/ftl v0.0.0-00010101000000-000000000000
debug:other: Compiling
trace:other: exec: cd /Users/alec/dev/ftl/buildengine/testdata/projects/other/_ftl/go/main && go build -o ../../main .
    /Users/alec/dev/ftl/buildengine/build_go_test.go:342: Expected values to be equal:
         	"github.com/TBD54566975/ftl/go-runtime/ftl/reflection"
         	"github.com/TBD54566975/ftl/go-runtime/server"
         
        -	"ftl/another"
         	"ftl/other"
         )
         
         func init() {
         	reflection.Register(
        -		reflection.WithSumType[another.SecondTypeEnum](
        -			*new(another.One),
        -			*new(another.Two),
        -		),
        -		reflection.WithSumType[another.TypeEnum](
        -			*new(another.A),
        -			*new(another.B),
        -		),
         		reflection.WithSumType[other.SecondTypeEnum](
         			*new(other.A),
         			*new(other.B),
```

**Comment by alecthomas** [comment]
I also just realised that this shouldn't be registering types for `another` at all, as this is the main entry point for the `other` package.

**Comment by alecthomas** [comment]
This is now seemingly failing a great deal of the time.

**Comment by alecthomas** [comment]
This is not really "fixed", just ignored?

**Comment by wesbillman** [comment]
yeah sorry I'm not 100% sure either. Other than we had the wrong test code in there I think since it had `another` in it. if I'm reading it correctly. So it seemed like it would just fail every time I ran it. Correcting the code make it pass. So my theory was that maybe the code was wrong for a bit, but worked sometimes before we added code to delete the `_ftl` folder completely during builds.

Happy to revert the change though. Or we can see if it's still flaky for people with these updates.
